src/app/game-state/log.service.ts

```diff
@@ -135,53 +127,38 @@ export class LogService {
   }
 
   updateLogTopics(){
-    const logs: Log[][] = [];
+    const logs: Log[] = [];
 
-    if (this.logTopics.includes('COMBAT')){
-      this.newCombat = "";
-      logs.push([...this.combatLog]);
-    }
     if (this.logTopics.includes('STORY')){
       this.newStory = "";
-      logs.push([...this.storyLog]);
-      }
+      logs.push(...this.storyLog);
+    }
     if (this.logTopics.includes('EVENT')){
       this.newEvents = "";
-      logs.push([...this.eventLog]);
+      this.eventLog = this.eventLog.slice(-300)
+      logs.push(...this.eventLog);
+    }
+    if (this.logTopics.includes('COMBAT')){
+      this.newCombat = "";
+      this.combatLog = this.combatLog.slice(-300)
+      logs.push(...this.combatLog);
     }
     if (this.logTopics.includes('CRAFTING')){
       this.newCrafting = "";
-      logs.push([...this.craftingLog]);
+      this.craftingLog = this.craftingLog.slice(-300)
+      logs.push(...this.craftingLog);
     }
     if (this.logTopics.includes('FOLLOWER')){
       this.newFollower = "";
-      logs.push([...this.followerLog]);
+      this.followerLog = this.followerLog.slice(-300)
+      logs.push(...this.followerLog);
     }
 
-    this.currentLog = [];
     if (logs.length === 0){
       return;
     }
-    //@ts-ignore
-    const isEmpty = a => Array.isArray(a) && a.every(isEmpty);
-    while(!isEmpty(logs)){
-      // figure out the oldest log entry and add it to the currentLog until everything is added
-      let latestTimestamp = Number.MAX_VALUE;
-      let latestLog: Log[] = logs[0];
-      for (let index = 0; index < logs.length; index++){
-        const loopLog = logs[index];
-        if (loopLog.length === 0){
-          continue;
-        }
-        const timestamp = loopLog[loopLog.length - 1].timestamp || 0;
-        if (timestamp < latestTimestamp){
-          latestTimestamp = timestamp;
-          latestLog = loopLog;
-        }
-      }
-      this.addToCurrentLog(latestLog[latestLog.length - 1]);
-      latestLog.splice(latestLog.length - 1, 1);
-    }
+    logs.sort((a,b) => b.timestamp - a.timestamp);
+    this.currentLog = logs.slice(0,299);
```

Comment:
Instead of emptying currentLog, just give it the most recent slice and let the garbage collection sweep the old one.